src/symbolize/libbacktrace.rs

```diff
@@ -400,8 +400,17 @@ unsafe fn init_state() -> *mut bt::backtrace_state {
                     Ok(&buf[..(len + 1) as usize])
                 }
             }
-
-
+        } else if #[cfg(target_os = "vxworks")] {
+            unsafe fn load_filename() -> *const libc::c_char {
+                use libc;
+                use core::mem;
+                let mut rtp_desc : libc::RTP_DESC = mem::zeroed();
+                if (libc::rtpInfoGet(0, &mut rtp_desc as *mut libc::RTP_DESC) == 0){
+                    rtp_desc.pathName.as_mut_ptr() as *mut libc::c_char
```

Comment:
This looks wrong. The string is copied into the rtp_desc.pathName array. This is a piece of rtp_desc,  which is a local variable in this function. So I think we're returning a wild pointer here.

Other implementations of load_filename use a static buffer to avoid this. It appears from the usage that we can rely on this function only being called once.

Reply:
I am having trouble in adding "static" to rtp_desc.
Do you think it is a good idea to introduce a new function in vxWorks that just return rtp executable path? and if we name the new function as "getexecname", we even don't need to add any code in this file.